src/states/node.rs

```diff
@@ -1736,21 +1727,42 @@ impl Node {
     }
 
     // Received by Y; From X -> Y
-    // Context: a node is joining our section. Sends `AcceptAsCandidate` to our section. If the
-    // network is unbalanced, sends `ExpectCandidate` on to a section with a shorter prefix.
+    // Context: a node is joining our section. Vote `ExpectCandidate`.
     fn handle_expect_candidate(
         &mut self,
-        old_pub_id: PublicId,
+        old_public_id: PublicId,
         old_client_auth: Authority<XorName>,
-        relocation_dst: Authority<XorName>,
+        dst_name: XorName,
         message_id: MessageId,
     ) -> Result<(), RoutingError> {
-        self.remove_expired_peers();
+        if !self.chain.is_member() {
+            return Ok(());
+        }
+
+        self.vote_for_event(NetworkEvent::ExpectCandidate(ExpectCandidatePayload {
+            old_public_id,
+            old_client_auth,
+            dst_name,
+            message_id,
+        }));
+        Ok(())
+    }
 
-        if old_pub_id == *self.full_id.public_id() {
-            return Ok(()); // This is a delayed message belonging to our own relocate request.
+    // Handles an accumulated `ExpectCandidate` event.
+    // Context: a node is joining our section. Sends the node our section. If the
+    // network is unbalanced, sends `ExpectCandidate` on to a section with a shorter prefix.
+    fn handle_expect_candidate_event(
+        &mut self,
+        vote: ExpectCandidatePayload,
```

Comment:
Why not borrow `ExpectCandidatePayload` instead of moving it here?

Reply:
We then move things out. Removed the unnecessary clone at the call site.
Though all the members are copy... But that should be enough to address this comment I think.